recovery.go

```diff
@@ -0,0 +1,83 @@
+package handlers
+
+import (
+	"log"
+	"net/http"
+	"runtime/debug"
+)
+
+type recoveryHandler struct {
+	handler    http.Handler
+	logger     *log.Logger
+	printStack bool
+}
+
+// Option provides a functional approach to define
+// configuration for a handler; such as setting the logging
+// whether or not to print strack traces on panic.
+type Option func(http.Handler)
```

Comment:
This should probably be `RecoveryOption` as not to conflate with the other handlers in the package.